pgjdbc/src/main/java/org/postgresql/core/v2/ProtocolConnectionImpl.java

```diff
@@ -197,6 +209,15 @@ synchronized void addWarning(SQLWarning newWarning) {
 
   synchronized void addNotification(PGNotification notification) {
     notifications.add(notification);
```

Comment:
On 2 June 2016 at 11:27, David Johnston <notifications@github.com> wrote:

> In pgjdbc/src/main/java/org/postgresql/core/v2/ProtocolConnectionImpl.java
> <https://github.com/pgjdbc/pgjdbc/pull/579#discussion_r65561786>:
>
> > @@ -197,6 +209,15 @@ synchronized void addWarning(SQLWarning newWarning) {
> >
> >    synchronized void addNotification(PGNotification notification) {
> >      notifications.add(notification);
>
> I'd be inclined to make it an either/or proposition. If you establish a
> listener we don't queue up the notifications - only raise them via event.
> If no listeners are present we continue with the present logic.
>

That makes sense. Thinking about this some more, I think I'd like to
deprecate the notifications.

Dave Cramer

> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pgjdbc/pgjdbc/pull/579/files/229d10adf6686805004491a0d94de076d8e1a3ea#r65561786>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAYz9u80w6_Oth2mCOyL_vhqMTkx66EKks5qHvZVgaJpZM4IsnG2>
> .
>